modules/connectivity/ParticipantConnectionStatus.js

```diff
@@ -380,18 +455,117 @@ export default class ParticipantConnectionStatus {
+    }
+
+    /**
+     * On change in Last N set check all leaving and entering participants to
+     * change their corresponding statuses.
+     *
+     * @param {Array<string>} leavingLastN array of ids leaving lastN.
+     * @param {Array<string>} enteringLastN array of ids entering lastN.
+     * @private
+     */
+    _onLastNChanged(leavingLastN, enteringLastN) {
+        if (leavingLastN) {
+            leavingLastN.forEach(id => {
+                delete this.enteredLastNTimestamp[id];
+                this._clearRestoringTimer(id);
+                this.figureOutConnectionStatus(id);
+            });
+        }
+        if (enteringLastN) {
+            enteringLastN.forEach(id => {
+                // store the timestamp this id is entering lastN
+                this.enteredLastNTimestamp[id] = Date.now();
+
+                this.figureOutConnectionStatus(id);
+            });
+        }
+    }
+
+    /**
+     * Clears the restoring timer for participant's video track and the
+     * timestamp for entering lastN.
+     *
+     * @param {string} participantId the id of the conference participant which
+     * is the same as the Colibri endpoint ID of the video channel allocated for
+     * the user on the videobridge.
+     */
+    _clearRestoringTimer(participantId) {
+        if (this.restoringTimers[participantId]) {
+            clearTimeout(this.restoringTimers[participantId]);
+            delete this.restoringTimers[participantId];
+        }
+    }
+
+    /**
+     * Checks whether a track had stayed enough in restoring state, compares
+     * current time and the time the track entered in lastN. If it hasn't
+     * timedout and there is no timer added, add new timer in order to give it
+     * more time to become active or mark it as interrupted on next check.
+     *
+     * @param {string} participantId the id of the conference participant which
+     * is the same as the Colibri endpoint ID of the video channel allocated for
+     * the user on the videobridge.
+     * @returns {boolean} <tt>true</tt> if the track was in restoring state
+     * more than the timeout ({@link DEFAULT_RESTORING_TIMEOUT}.) in order to
+     * set its status to interrupted.
+     * @private
+     */
+    _isRestoringTimedout(participantId) {
+        const enteredLastNTimestamp
+            = this.enteredLastNTimestamp[participantId];
+
+        if (typeof enteredLastNTimestamp === 'number'
+            && (Date.now() - enteredLastNTimestamp)
+            >= DEFAULT_RESTORING_TIMEOUT) {
+            return true;
+        }
+
+        // still haven't reached timeout, if there is no timer scheduled,
+        // schedule one so we can track the restoring state and change it after
+        // reaching the timeout
+        if (!this.restoringTimers[participantId]) {
+            this.restoringTimers[participantId] = setTimeout(
+                this.figureOutConnectionStatus.bind(this, participantId),
```

Comment:
that's one of the reasons why arrow functions exist, see the first paragraph here: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions

If that doesn't convince you: MORE NEWER MORE BETTER :-P